headers/xdp/xsk.h

```diff
@@ -201,8 +201,35 @@ struct xsk_umem_config {
 	__u32 frame_size;
 	__u32 frame_headroom;
 	__u32 flags;
+	__u32 tx_metadata_len;
```

Comment:
Adding to the usr_config struct kinda defeats the purpose of the OPTS usage. libxdp still has no way of knowing whether the caller set this field or not.

Instead, make all the members of xsk_umem_config individual members of xsk_umem_opts, and read each one with the OPTS_GET macro.